src/core.rs

```diff
@@ -311,9 +315,11 @@ pub enum Event {
 /**
     ToxOptions provides options that tox will be initalized with.
 */
-#[derive(Copy, Clone)]
+
+//#[derive(Clone, Copy)]
 pub struct ToxOptions {
-    raw: ll::Tox_Options
+    raw: ll::Tox_Options,
+    sk_ptr: Option<*mut SecretKey>,
```

Comment:
Why `*mut SecretKey`?

Reply:
We have to keep sk somewhere in a pin location, I chose to keep it in heap.
Assume ToxOptions is moved (like from one function into another), but the pointer to savedata_data should stay the same.

Reply:
`Pin` is available in stable rust. Beside that, it's probably better to use `UnsafeCell` instead of `*mut`.

Reply:
> Pin is available in stable rust.

I don't need to use Pin.

> Beside that, it's probably better to use UnsafeCell instead of *mut.

Why? You use raw pointers everywhere, what's the difference?

Reply:
Rstox indeed has some historical decisions.

I'm not a fan of tricks with raw pointers, but if you like it then merge it.

Reply:
> I'm not a fan of tricks with raw pointers

Me neither.

> better to use UnsafeCell instead of *mut.

I don't know how to use them.